fluentd/install.sls

```diff
@@ -0,0 +1,65 @@
+{% from "fluentd/map.jinja" import fluentd, fluentd_service with context %}
+
+install_fluentd_dependencies:
+  pkg.installed:
+    - pkgs: {{ fluentd.pkgs }}
+    - update: True
+{% endif %}
+
+install_fluentd_gem:
+  gem.installed:
+    - name: fluentd
+    {% if fluentd.version %}
+    - version: {{ fluentd.version }}
+    {% endif %}
+
+create_fluentd_user:
+  user.present:
+    - name: {{ fluentd.user }}
+    - createhome: False
+
+create_fluentd_group:
+  group.present:
+    - name: {{ fluentd.group }}
+    - addusers:
+        - {{ fluentd.user }}
+
+configure_fluentd:
+  file.managed:
+    - name: /etc/fluent/fluent.conf
+    - source: salt://fluentd/templates/fluent.conf
+    - makedirs: True
+    - user: {{ fluentd.user }}
+    - group: {{ fluentd.group }}
+    - template: jinja
+    - context:
+        log_level: {{ fluentd.global_log_level }}
+
+make_fluent_config_directory:
+  file.directory:
+    - name: /etc/fluent/fluent.d/
+    - makedirs: True
+    - user: {{ fluentd.user }}
+    - group: {{ fluentd.group }}
```

Comment:
This should also have the parameter
```
recurse:
  - user
  - group
```
to ensure proper ownership of files and subdirectories